caddyconfig/httpcaddyfile/options_test.go

```diff
@@ -0,0 +1,92 @@
+package httpcaddyfile
+
+import (
+	"testing"
+
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
+	_ "github.com/caddyserver/caddy/v2/modules/logging"
+)
+
+func TestGlobalLogOptionSyntax(t *testing.T) {
+	for i, tc := range []struct {
+		input       string
+		output      string
+		expectError bool
+	}{
+		{
+			input: `{
+				log
+			}
+			`,
+			output:      `{}`,
+			expectError: false,
+		},
+		{
+			input: `{
+				log {
+					output file foo.log
+				}
+			}
+			`,
+			output:      `{"logging":{"logs":{"default":{"writer":{"filename":"foo.log","output":"file"}}}}}`,
+			expectError: false,
+		},
+		{
+			input: `{
+				log {
+					format filter {
+						wrap console
+						fields {
+							common_log delete
+							request>remote_addr ip_mask {
+								ipv4 24
+								ipv6 32
+							}
+						}
+					}
+				}
+			}
+			`,
+			output:      `{"logging":{"logs":{"default":{"encoder":{"fields":{"common_log":{"filter":"delete"},"request\u003eremote_addr":{"filter":"ip_mask","ipv4_cidr":24,"ipv6_cidr":32}},"format":"filter","wrap":{"format":"console"}}}}}}`,
+			expectError: false,
+		},
+		{
+			input: `{
+				log {
+					output file foo.log
+				}
+				log default {
+					format json
+				}
+			}
+			`,
+			output:      `{"logging":{"logs":{"default":{"encoder":{"format":"json"}}}}}`,
+			expectError: false,
```

Comment:
I'm confused about this one - shouldn't this be an error? I don't think it's valid to have a `log` without a name in globals.

Reply:
I was going off your comment here about how the name could be optional: https://github.com/caddyserver/caddy/issues/3958#issuecomment-756776300

This is a bit of a strange case because the second block overwrites the first one.

I like the idea of making it name required field, explicit in all cases for the global option. I'll make that change.

Reply:
Oh I see, well in the case yeah I think it's okay that omitting the name uses `default`, but it should be an error if the same name is used (or implied) more than once. Overriding is non-apparent behaviour and is almost always going to be misconfiguration.

Reply:
Got it, makes sense to me, I'll add back the allowance for making it optional.